ez/ez.go

```diff
@@ -128,8 +137,22 @@ func ConfigFileEnvFlag(ctx context.Context, cfg ConfigWithConfigPath, df Decoder
 		defer cancel()
 		ctx = configCtx
 	}
+	// OnWatchedError is never called from this goroutine, so it can be
+	// unbuffered without deadlocking.
+	blankErrCh := make(chan error, 1)
```

Comment:
If it can be unbuffered, then why have buffer size == 1?

Reply:
I think because we don't want to send the error to that channel every time. We only want to send it to the channel at startup and then call the `option.onWatchedError` callback for subsequent errors. Buffering ensures that we hit the default case after the initial call.

Reply:
I've extended the comment to explain why it's buffered.

Reply:
Because github's review UI is so terrible, here are the new contents of the comment above this channel:
```go
	// OnWatchedError is never called from this goroutine, so it can be
	// unbuffered without deadlocking.
	//
	// However, it is buffered to avoid a race where the non-blocking send
	// in the callback happens before the select statement at the bottom of
	// this function starts. If this weren't buffered, the send would fall
	// through to the delegated error-handler, and the select statement
	// would block until a new config version (or error) was created
	// (assuming this is a watching-mode) or possibly forever.

```